torch/csrc/jit/python/init.cpp

```diff
@@ -1879,16 +1882,21 @@ void initJITBindings(PyObject* module) {
               }),
           py::call_guard<py::gil_scoped_release>());
   m.def("_is_alias_of", [](const py::object& self, const py::object& other) {
-    if (isEmptyContainer(self) || isEmptyContainer(other)) {
+    c10::optional<IValue> self_value = toTypeInferredIValueOptional(self);
+    c10::optional<IValue> other_value = toTypeInferredIValueOptional(other);
+
+    if (!self_value || !other_value) {
```

Comment:
maybe also add a comment here (e.g. "only return true if we can be certain that they are aliasing")